### PR TITLE
CHIT rewards backend part 1 - streaks

### PR DESCRIPTION
**Pull request by megrogan**

### CODE REVIEW

File: backend/canisters/local_user_index/api/src/lib.rs [hunk 1]
```diff
@@ -162,3 +163,11 @@ pub struct GlobalUser {
     pub is_platform_moderator: bool,
     pub diamond_membership_expires_at: Option<TimestampMillis>,
 }
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct ChitEarned {
+    pub user_id: UserId,
+    pub amount: i32,
```
Comment (hpeebles) [inline backend/canisters/local_user_index/api/src/lib.rs:170]: `i32` in order to have deductions?

Comment (megrogan) [reply backend/canisters/local_user_index/api/src/lib.rs:170]: yes

File: backend/canisters/user_index/impl/src/queries/current_user.rs [hunk 2]
```diff
@@ -43,6 +44,9 @@ fn current_user_impl(state: &RuntimeState) -> Response {
             diamond_membership_details: u.diamond_membership_details.hydrate(now),
             diamond_membership_status: u.diamond_membership_details.status_full(now),
             moderation_flags_enabled: u.moderation_flags_enabled,
+            chit_balance: u.chit_balance,
+            streak: u.streak.days(now),
+            next_daily_claim: if u.streak.can_claim(now) { midnight_today(now) } else { midnight_tomorrow(now) },
```
Comment (hpeebles) [inline backend/canisters/user_index/impl/src/queries/current_user.rs:49]: Surely if they can claim now then the next daily claim is yesterday midnight? Or simply just now?

Comment (megrogan) [reply backend/canisters/user_index/impl/src/queries/current_user.rs]: renamed

**Review by hpeebles — APPROVED** [review]

**Pull request merged**